File: comparisons/shapes.java
import java.util.Scanner;

public class ShapeCalcs {
	public static void main(String arge[]) {
		Scanner in = new Scanner(System.in);
		
		System.out.println("Enter rectangle width: ");
		int width = in.nextInt();
		
		System.out.println("Enter rectangle height: ");
		int height = in.nextInt();
		
		String choice = "";
		while (true) {			
			System.out.println("Enter calculation: (Perimeter/Area: )");
			choice = in.next();
			
			System.out.println((choice.equalsIgnoreCase("perimeter") ?
					"Perimeter: " + peri(width, height) : "Area: " + area(width, height)));
			
			System.out.println("Display both? [y/n]");
			choice = in.next();
			
			if (choice.equalsIgnoreCase("y")) {
				calcAll(width, height);
			}
			
			System.out.println("Done.\n");
		}
	}

	private static int peri(int width, int height) {
		return 2*width + 2*height;
	}

	private static int area(int width, int height) {
		return width*height;
	}

	private static void calcAll(int width, int height) {
		int localPerm = peri(width, height);
		int localArea = area(width, height);
		
		System.out.println("Perimeter: " + localPerm);
		System.out.println("Area: " + localArea);
	}
}

File: comparisons/shapes.cs
using System;

class ShapeCalcs {
    static void Main() {

        Console.Write("Enter rectangle width: ");
        int width = int.Parse(Console.ReadLine());
        Console.Write("Enter rectangle height: ");
        int height = int.Parse(Console.ReadLine());

        string choice = "";
        while (true) {
            Console.Write("Enter calculation: (Perimeter/Area: )");
            choice = Console.ReadLine();

            if (choice.ToLower() == "perimeter") {
                Console.WriteLine("Perimeter: " + Peri(width, height));
            }
            else if (choice.ToLower() == "area") {
                Console.WriteLine("Area: " + Area(width, height));
            }

            Console.Write("Display both? [y/n]: ");
            choice = Console.ReadLine();

            if (choice.ToLower() == "y") {
                CalcAll(width, height);
            }

            Console.WriteLine("Done.\n");
        }
    }

    private static int Peri(int width, int height) {
        return 2 * width + 2 * height;
    }

    private static int Area(int width, int height) {
        return width * height;
    }

    private static void CalcAll(int width, int height) {
        int localPerm = Peri(width, height);
        int localArea = Area(width, height);

        Console.WriteLine("Perimeter: " + localPerm);
        Console.WriteLine("Area: " + localArea);
    }
}